### PR TITLE
Set gpuNum as optional

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

**Comment by liuzhe-lz** [comment]
It seems GPUs are still available when `CUDA_VISIBLE_DEVICES` is set to 0, while Tensorflow will only use 1 GPU in this case.
Can we use this as the default setting?

File: src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts [hunk 1]
```diff
@@ -601,12 +601,15 @@ class RemoteMachineTrainingService implements TrainingService {
         let command: string;
         // Set CUDA_VISIBLE_DEVICES environment variable based on cuda_visible_device
         // If no valid cuda_visible_device is defined, set CUDA_VISIBLE_DEVICES to empty string to hide GPU device
-        if (typeof cuda_visible_device === 'string' && cuda_visible_device.length > 0) {
-            command = `CUDA_VISIBLE_DEVICES=${cuda_visible_device} ${this.trialConfig.command}`;
+        if (this.trialConfig.gpuNum === undefined) {
```
Comment (yds05) [inline src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts]: Add comments to explain if gpuNum is undefined we will not set CUDA_VISIBLE_DEVICE?

File: docs/en_US/Tutorial/ExperimentConfig.md [hunk 2]
```diff
@@ -266,7 +266,7 @@ machineList:
 
   * __gpuNum__
 
-      __gpuNum__ specifies the gpu number to run the tuner process. The value of this field should be a positive number.
+      __gpuNum__ specifies the gpu number to run the tuner process. The value of this field should be a positive number. If the field is not set, NNI will not set `CUDA_VISIBLE_DEVICE` in script, and will not manage gpu resource.
```
Comment (QuanluZhang) [inline docs/en_US/Tutorial/ExperimentConfig.md]: => in script (that is, will not control the visibility of GPUs on trial command through `CUDA_VISIBLE_DEVICES`), and ...

CUDA_VISIBLE_DEVICE => CUDA_VISIBLE_DEVICES

File: src/nni_manager/training_service/remote_machine/gpuScheduler.ts [hunk 3]
```diff
@@ -49,6 +49,9 @@ export class GPUScheduler {
      * @param requiredGPUNum required GPU number
      */
     public scheduleMachine(requiredGPUNum: number, trialJobDetail : RemoteMachineTrialJobDetail) : RemoteMachineScheduleResult {
```
Comment (QuanluZhang) [inline src/nni_manager/training_service/remote_machine/gpuScheduler.ts]: is the type `number` for requiredGPUNum still correct?

Comment (QuanluZhang) [inline src/nni_manager/training_service/remote_machine/gpuScheduler.ts]: number -> number | undefined ?

**Comment by QuanluZhang** [comment]
what is the behavior in kubeflow/pai mode when gpuNum is not specified?

File: docs/en_US/Tutorial/ExperimentConfig.md (hunk 2 above)
Comment (SparkSnail) [reply docs/en_US/Tutorial/ExperimentConfig.md]: fixed.

File: src/nni_manager/training_service/remote_machine/gpuScheduler.ts (hunk 3 above)
Comment (SparkSnail) [reply src/nni_manager/training_service/remote_machine/gpuScheduler.ts]: fixed.

**Comment by SparkSnail** [comment]
> what is the behavior in kubeflow/pai mode when gpuNum is not specified?

In kubeflow/pai mode, the gpuNum field is required, since the gpu schedule policy does not relay on `CUDA_VISIBLE_DEVICES`.

**Comment by SparkSnail** [comment]
> It seems GPUs are still available when `CUDA_VISIBLE_DEVICES` is set to 0, while Tensorflow will only use one GPU in this case.
> Can we use this as default setting?
> 
> Why did we use `CUDA_VISIBLE_DEVICES=-1`? @QuanluZhang

 `CUDA_VISIBLE_DEVICES=-1` is set to block all of gpus.

**Pull request merged**